src/span_ext.rs

```diff
@@ -207,4 +226,15 @@ impl OpenTelemetrySpanExt for tracing::Span {
             }
         });
     }
+
+    fn set_status(&self, status: Status) {
+        self.with_subscriber(move |(id, subscriber)| {
+            let mut status = Some(status);
```

Comment:
Why do you wrap the status in `Option` here?

Reply:
The answer is quite complicated and I was confused myself for a bit. This forum post helped me understand: https://users.rust-lang.org/t/cant-move-a-variable-out-of-fnmut-closure/41497/3

It boils down to the `status` variable gets moved twice. Once in the lambda of `with_subscriber` and again with `with_context`. The `with_context` signature is the following:

```
pub(crate) fn with_context(
        &self,
        dispatch: &tracing::Dispatch,
        id: &span::Id,
        mut f: impl FnMut(&mut OtelData, &dyn PreSampledTracer),
    )
```

`FnMut` is allowed to be called multiple times, and could in theory move `status` multiple times which results in the following error:

```
 fn set_status(&self, status: Status) {
    |                          ------ captured outer variable
...
234 |                 get_context.with_context(subscriber, id, move |builder, _| {
    |                                                          ----------------- captured by this `FnMut` closure
235 |                     builder.builder.status = status;
    |                                              ^^^^^^ move occurs because `status` has type `Status`, which does not implement the `Copy` trait
```

Thus it is telling us that `status` is no longer valid after the first move.

There are few ways to address this.

1 - call `status.clone()`. This will work at the cost of a clone operation because the lambda will have its own independent copy of the status.

2. Put the status into an `Option<T>` and use `take()`. The only reason I did this was there is prior art of this earlier in the file. Notice lines 213-214: 

```
let mut key = Some(key.into());
let mut value = Some(value.into());
```
It is using the `Option<T>` approach to address the same issue, hence to be consistent I used this.

3. See if we can add/change a `with_context` function that accepts a `FnOnce` I tried doing this, but the change kind of spreads through the entire library.

Perhaps there is something I am missing and I am open to other options

Reply:
That's fair, thanks for the explanation/